src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +58,39 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+            IReadOnlyList<PackageSourceMappingSourceItem> existingSettingsLookup = GetPackageSourceMappingItems();
```

Comment:
Wouldn't this prevent this method from adding a *completely* new mapping?

Say you start a config with no mappings whatsoever. 
If you use the save method, you wouldn't update anything? 

fwiw, the `Remove` and `AddOrUpdatePackageSourceMappingSourceItem` methods already have logic that allows you to initialize a config.

Reply:
This change was in response to PR feedback asking for the null check to be generalized. I think it was designed appropriately for "completely new" scenario before, but now it has a new bug.

Reply:
The user could still initialize a new mapping since GetPackageSourceMappingItems() will never return null (it will just return an empty list if there are no mappings). I did remove the if statement, though, because it was redundant and made a test that adds a new mapping if there are none to begin with.

Reply:
Resolving this thread.

Feel free to resolve threads that have feedback that's been fully addressed :)